docs/reference/index.rst

```diff
@@ -7,7 +7,14 @@
 .. toctree::
    :hidden:
    :maxdepth: 2
-   :caption: Introduction
+   :caption: Reference
+
+   src/reference/syntax/index.rst
+
+.. toctree::
+   :hidden:
+   :maxdepth: 2
+   :caption: Introduction (Old)
```

Comment:
I don't think you meant to merge this to master.
Should this go in the feature branch?